### PR TITLE
build(flatpak): more improvements to manifest

### PR DESCRIPTION
**Pull request by underlinejakez**
# Description
- remove node anchor used in the yml version
- remove gamemode module (freedesktop sdk 24.08 has already bundled)
- disable portal-tests, vala binds and gtk3 backend of libportal
- make explicit the buildsystem of imagemagick

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Misc

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. 
Provide instructions so we can reproduce.
- [x] Building locally.

### CODE REVIEW

File: build-aux/com.usebottles.bottles.json [hunk 1]
```diff
@@ -402,4 +317,4 @@
             ]
         }
     ]
-}
\ No newline at end of file
+}
```
Comment (TheEvilSkeleton) [inline build-aux/com.usebottles.bottles.json:320]: This is unrelated

File: build-aux/com.usebottles.bottles.json [hunk 2]
```diff
@@ -247,6 +159,7 @@
         },
         {
             "name": "ImageMagick",
+            "buildsystem": "autotools",
```
Comment (TheEvilSkeleton) [inline build-aux/com.usebottles.bottles.json:162]: `autotools` is implicitly used, so we don't need to explicitly set it

**Review by TheEvilSkeleton — CHANGES_REQUESTED** [review]

File: build-aux/com.usebottles.bottles.json (hunk 2 above)
Comment (underlinejakez) [reply build-aux/com.usebottles.bottles.json:162]: it's just to be more clear, i couldn't find in the flatpak builder docs about being the default for buildsystem
also the module right above is explicitly set as autotools

File: build-aux/com.usebottles.bottles.json (hunk 1 above)
Comment (underlinejakez) [reply build-aux/com.usebottles.bottles.json]: its automatic formatting, idk how to not do, not a big deal but i can look into if needed

**Comment by underlinejakez** [comment]
forgot the vulkan-headers change in the commit but i guess its fine

File: build-aux/com.usebottles.bottles.json (hunk 1 above)
Comment (TheEvilSkeleton) [reply build-aux/com.usebottles.bottles.json]: I guess it depends on what @mirkobrombin thinks. Personally, to avoid polluting `git blame`, I prefer not to have unrelated changes

**Review by TheEvilSkeleton — APPROVED** [review]
Other than https://github.com/bottlesdevs/Bottles/pull/3585#discussion_r1879008279, everything looks fine to me

File: build-aux/com.usebottles.bottles.json (hunk 1 above)
Comment (mirkobrombin) [reply build-aux/com.usebottles.bottles.json]: We can keep it for this time
